dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/reservedvalue/hibernate/HibernateReservedValueStore.java

```diff
@@ -187,6 +179,19 @@ public boolean isReserved( String ownerObject, String ownerUID, String value )
             .isEmpty();
     }
 
+    @Override
+    public void removeUsedOrExpiredReservations()
+    {
+        String deleteQuery = "DELETE FROM ReservedValue r WHERE r.expiryDate < :now OR r.value in (" +
+            "SELECT teav.plainValue from TrackedEntityAttributeValue teav JOIN teav.attribute tea " +
+            "WHERE r.ownerUid = tea.uid and r.value = teav.plainValue" +
```

Comment:
I think you can use the JPA built-in `CURRENT_TIMESTAMP` keyword instead of a parameter @lucaCambi77 .

https://www.objectdb.com/java/jpa/query/jpql/date

Reply:
ok thanks i'll try

Reply:
I didn't test to check if it's exactly the same, however the following could be faster:

```SQL
DELETE
FROM ReservedValue r
WHERE r.expiryDate < CURRENT_TIMESTAMP
   OR EXISTS(
        SELECT 1
        FROM TrackedEntityAttributeValue teav
                 JOIN teav.attribute tea
        WHERE r.ownerUid = tea.uid
          AND r.value = teav.plainValue
    )
```

Also we should keep same case for SQL keywords, either all uppercase or all lowercase for better readability

Reply:
@gnespolino Looking at execution plan apparently we don't gain much so i'll leave it as it is, but we can monitor it and change it in the future thanks